lib/http-assertions.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * @param {Response} res
+ * @param {string} [errorMsg]
+ */
+export async function assertOkResponse (res, errorMsg) {
+  if (res.ok) return
+
+  let body
+  try {
+    body = await res.text()
+  } catch {}
```

Comment:
Please store the error (or at least the error message) in `body`. Otherwise, the error message created below would be difficult to work with.